src/workerd/api/worker-rpc.c++

```diff
@@ -2,19 +2,401 @@
+  if (caughtException) {
+    // The remote Worker threw an exception, so we should throw on the client side too.
+    js.throwException(deserialized);
+  }
+
+  KJ_REQUIRE(customEvent.outcome == EventOutcome::OK, "got unexpected event outcome for WorkerRpc");
+  return jsg::Value(js.v8Isolate, kj::mv(deserialized));
+}
+
 kj::Maybe<jsg::JsValue> WorkerRpc::getNamed(jsg::Lock& js, kj::StringPtr name) {
-    // TODO(soon): The server side PR will fill this function with the RPC call and other stuff,
-    //             this is just a temporary placeholder for testing.
-    return jsg::JsValue(js.wrapReturningFunction(js.v8Context(),
-        [](jsg::Lock& js, const v8::FunctionCallbackInfo<v8::Value>& args) {
-          return js.error("WorkerRpc is unimplemented");
+  // Named intercept is enabled, this means we won't default to legacy behavior.
+  // The return value of the function is a promise that resolves once the remote returns the result
+  // of the RPC call.
+  return jsg::JsValue(js.wrapReturningFunction(js.v8Context(), [this, methodName=kj::str(name)]
+      (jsg::Lock& js, const v8::FunctionCallbackInfo<v8::Value>& args) {
+        auto& ioContext = IoContext::current();
+        // Wait for the RPC to resolve and then process the result.
+        return js.wrapSimplePromise(ioContext.awaitIo(js, sendWorkerRpc(js, methodName, args),
+            [this] (jsg::Lock& js, WorkerRpc::ResponseAndResult result) -> jsg::Value {
+          return handleRpcResponse(js, kj::mv(result));
+        }));
+      }
+  ));
+}
+
+// The capability that lets us call remote methods over RPC.
+// The client capability is dropped after each callRequest().
+class JsRpcTargetImpl final : public rpc::JsRpcTarget::Server {
+public:
+  JsRpcTargetImpl(
+      kj::Own<kj::PromiseFulfiller<void>> callFulfiller,
+      IoContext& ctx,
+      kj::Maybe<kj::StringPtr> entrypointName)
+      : callFulfiller(kj::mv(callFulfiller)), ctx(ctx), entrypointName(entrypointName) {}
+
+  // Handles the delivery of JS RPC method calls.
+  kj::Promise<void> call(CallContext callContext) override {
+    auto methodName = kj::heapString(callContext.getParams().getMethodName());
+    auto serializedArgs = callContext.getParams().getSerializedArgs().getV8Serialized().asBytes();
+
+    // Try to execute the requested method.
+    try {
+      co_await ctx.run(kj::coCapture(
+          [this, methodName=kj::mv(methodName), serializedArgs=kj::mv(serializedArgs),
+           entrypointName=entrypointName, callContext] (Worker::Lock& lock) mutable ->
+              kj::Promise<void> {
+
+        jsg::Lock& js = lock;
+        if (!FeatureFlags::get(js).getJsRpc()) {
+          // JS RPC is not enabled on the server side, we cannot call any methods.
+          auto rpcDisabled =
+              kj::str("The receiving Worker does not allow its methods to be called over RPC."_kj);
+          // We need to save the JsValue here, otherwise the 'windows-debug' build complains.
```

Comment:
I saw your comment elsewhere that you were getting an exception about storing a JsValue on the heap.

The problem is that you're in a coroutine here, and the compiler is therefore free to store local variables and even temporaries on the coroutine frame instead of on the stack. The coroutine frame lives on the heap. But `JsValue` is not allowed to be stored on the heap, only on the stack. This is because a `JsValue` is invalidated as soon as we exit the innermost `v8::HandleScope`, so we want to make sure people don't try to store them long-term.

In a coroutine, whether the compiler decides to store any particular value on the stack vs. the coroutine frame is completely up to the compiler, and the decision can and does change between compiler versions, optimization levels, and probably even based on seemingly unrelated changes to the function. Because of this, you cannot use `JsValue` inside a coroutine.

But it was dangerous to use anyway, per my previous comments. After the first `co_await`, any `JsValue` is invalid, and the isolate lock isn't even held anymore.

Refactoring this code to keep JS API stuff out of coroutines should solve the problem.